src/printer.js

```diff
@@ -1698,8 +1704,13 @@ function genericPrintNoParens(path, options, print) {
         "type ",
         path.call(print, "id"),
         path.call(print, "typeParameters"),
-        " = ",
-        path.call(print, "right"),
+        " =",
+        hasLeadingOwnLineComment(options.originalText, n.right)
+          ? indent(
```

Comment:
Okay, so this is a bit tricky, the comments are printed around the node. So from this code perspective, `printed` is `// comment \n | type` so we need to indent here and put a \n so that it goes to the next line.

If we try to put the indent inside the node, then it would look

```js
type X =
// comment
  | type
```

notice that the `// comment` is not indented because outside.

Now, if you don't have any comments, then you are not sure if the node is going to be expanded or not, so you can't force a hardline and want to let the child decide how to indent itself, if needs be.